TDSClient/TDSClient/TDS/Login7/TDSLogin7PacketData.cs

```diff
@@ -133,137 +85,603 @@ public uint Length
+        /// <param name="tdsVersion">TDS Version</param>
+        /// <param name="packetSize">Packet size</param>
+        /// <param name="clientProgVer">Client program version</param>
+        /// <param name="connectionID">Connection ID</param>
+        /// <param name="clientLCID">Client LCID</param>
+        /// <param name="clientID">Client ID</param>
+        public TDSLogin7PacketData(
+            string hostName,
+            string appName,
+            string server,
+            string database,
+            uint tdsVersion = 1946157060,
+            uint packetSize = 4096,
+            uint clientProgVer = 117440512,
+            uint connectionID = 0,
+            uint clientLCID = 1033,
+            byte[] clientID = null)
         {
-            if (optionName == null || data == null)
-            {
-                throw new ArgumentNullException();
-            }
+            TDSVersion = tdsVersion;
+            PacketSize = packetSize;
+            ClientProgVer = clientProgVer;
+            ClientPID = (uint)Process.GetCurrentProcess().Id;
+            ConnectionID = connectionID;
+            ClientLCID = clientLCID;
+
+            OptionFlags1 = new TDSLogin7OptionFlags1();
+            OptionFlags2 = new TDSLogin7OptionFlags2();
+            OptionFlags3 = new TDSLogin7OptionFlags3();
+            TypeFlags = new TDSLogin7TypeFlags();
+            Options = new List<TDSLogin7Option>();
 
-            if (this.Options.Where(opt => opt.Name == optionName).Any())
+            AddLogin7CommonOptions();
+
+            if (clientID == null)
             {
-                throw new Exception("Login7 option already set!");
+                ClientID = new byte[] { 0x00, 0x01, 0x02, 0x03, 0x04, 0x05 };
             }
-
-            if (optionName != "Password" && optionName != "ChangePassword")
+            else if (clientID.Length != 6)
             {
-                LoggingUtilities.WriteLog($"  Adding Login7 option {optionName} [{data}].");
+                throw new Exception("Invalid ClientID length!");
             }
             else
             {
-                LoggingUtilities.WriteLog($"  Adding Login7 option {optionName}.");
+                ClientID = clientID;
             }
 
-            var option = TDSLogin7OptionFactory.CreateOption(optionName, data);
+            LoggingUtilities.WriteLog($" Adding option HostName with value [{Environment.MachineName}]");
+            HostName = Environment.MachineName;
 
-            this.Options.Add(option);
+            LoggingUtilities.WriteLog($"  Adding option ApplicationName with value [{appName}]");
+            ApplicationName = hostName;
+
+            LoggingUtilities.WriteLog($"  Adding option ServerName with value [{server}]");
+            ServerName = server;
+
+            LoggingUtilities.WriteLog($"  Adding option Database with value [{database}]");
+            Database = database;
+
+            ClientTimeZone = 480;
```

Comment:
What's this used for? Should it be hardcoded?

Reply:
![image](https://github.com/bogdanis99/SQL-Connectivity-Checker/assets/161343098/01f36d30-7aee-4385-ab6d-953fc4115861)
I will set it to 0